### PR TITLE
gpgv: add a gpgv command

### PR DESCRIPTION
**Pull request by rminnich**
usage
gpgv [-v] keyring signature file

There doesn't seem to be a way to do this in the domain of the crypto
package, or did I miss something?

Signed-off-by: Ronald G. Minnich <rminnich@google.com>

### CODE REVIEW

File: cmds/gpgv/gpgv.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+	if k, err = os.Open(kn); err != nil {
+		log.Fatalf("%v", err)
+	}
+	kr, err := openpgp.ReadKeyRing(k)
+	if err != nil {
+		log.Printf("ReadKeyRing: %v, trying Armored", err)
+		if k, err = os.Open(kn); err != nil {
+			log.Fatalf("reopen KeyRing: %v", err)
+		}
```
Comment (hugelgupf) [inline cmds/gpgv/gpgv.go]: Why reopen the keyring? Why not just pass `k` to `openpgp.ReadArmoredKeyRing`?

File: cmds/gpgv/gpgv.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+		kr, err = openpgp.ReadArmoredKeyRing(k)
+		if err != nil {
+			log.Fatalf("ReadArmoredKeyRing: %v", err)
+		}
+		check = openpgp.CheckArmoredDetachedSignature
+	}
+
+	if s, err = os.Open(sn); err != nil {
+		log.Fatalf("%v")
```
Comment (hugelgupf) [inline cmds/gpgv/gpgv.go]:
```go
log.Fatalf("%v", err)
```

use go vet -- it'd tell you about things like this.

File: cmds/gpgv/gpgv.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	}
+
+	if *verbose {
+		debug = log.Printf
+	}
+
+	kn, sn, fn := flag.Args()[0], flag.Args()[1], flag.Args()[2]
+
+	if k, err = os.Open(kn); err != nil {
+		log.Fatalf("%v", err)
```
Comment (hugelgupf) [inline cmds/gpgv/gpgv.go]: Each Fatalf could have some more identifying information for the user. E.g.

```go
log.Fatalf("error opening keyring file: %v", err)
```

also below

File: cmds/gpgv/gpgv.go (hunk 1 above)
Comment (rminnich) [inline cmds/gpgv/gpgv.go]: Yeah, good point, I tried that and ... did not work. And it was not seekable. I gave up. I could read it into a []byte and make a reader for that but it ends up about the same amount of code so ...

File: cmds/gpgv/gpgv.go (hunk 2 above)
Comment (rminnich) [inline cmds/gpgv/gpgv.go]: see the travis run ...

ok  	github.com/u-root/u-root/cmds/which	0.011s	coverage: 62.5% of statements
cmds/gpgv/gpgv.go:58: missing argument for Fatalf("%v"): format reads arg 1, have only 0 args

travis catches this as we do a full go vet on the codebase on each commit, but thanks, good catch.

File: cmds/gpgv/gpgv.go (hunk 3 above)
Comment (rminnich) [inline cmds/gpgv/gpgv.go]: Will do, although I believe it prints the file name too. But, yes, I agree.

**Comment by rminnich** [comment]
LGTM

**Pull request closed without merging**